apps/ipfs-podcasting/docker-compose.yml

```diff
@@ -0,0 +1,22 @@
+version: "3.7"
+
+services:
+  web:
+    image: ipfspodcasting/podcastnode:v0.5
+    init: true
+    restart: on-failure
+    stop_grace_period: 1m
+    user: "1000:1000"
+    ports:
+      - 4001:4001
+      - 5001:5001
+      - 8675:8675
```

Comment:
What are each of these ports needed for?

Reply:
4001 if for IPFS communication. Is it ok that this isn't over tor? If the umbrel is firewalled (like mine), they'll need to allow incoming/outgoing tcp/udp to port 4001

8675 is for the App UI (1st screenshot)

5001 is for the IPFS UI (2nd & 3rd screenshot)

Reply:
Having two different UIs on two different ports could be confusing. When the user clicks on the app icon we'll have to choose which UI to show. I see you have a link to the IPFS UI but that must hardcode the port. It would break if the port ever changes in the future. At some point in the future these ports will likely be assigned dynamically by Umbrel. Maybe we could run bot on the same port? The app UI could be server from `/admin` and `/*` could be proxied through to IPFS potentially?

You can checkout the Agora app where @yzernik used the same technique as an example.

Reply:
Yes. It should work fine if they're behind NAT. I only mentioned port forwarding if they lock down all ports.

Without the IPFS interface (5001), you could run "ipfs swarm peers" in the docker shell to check/view peers.

When I first setup my umbrel I locked down everything except tor (9001), but found I needed to open 123 (NTP) and 80/443 to get bitcoin and lightning connections going.